### PR TITLE
Provide an API to extract batches on demand

### PR DESCRIPTION
**Pull request by MGelbana-Incorta**
## Status
**READY**

## Description
Provide an API to enable users to retrieve batches on Users, Groups, Events or Roles on demand. More information can be found in this [Issue](https://github.com/onelogin/onelogin-java-sdk/issues/10).

## Todos
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by MGelbana-Incorta** [comment]
I've just noticed that many of the changed lines are just reformatted. Should I revert those reformatting changes ?

**Comment by pitbulk** [comment]
Yes please

File: src/main/java/com/onelogin/sdk/model/Role.java [hunk 1]
```diff
@@ -4,24 +4,42 @@
...
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
```
Comment (pitbulk) [inline src/main/java/com/onelogin/sdk/model/Role.java]: What is that 31?

Comment (MGelbana-Incorta) [reply src/main/java/com/onelogin/sdk/model/Role.java]: It's the auto generated hashCode function implementation provided by Eclipse. Should I implement it in any other way ?

**Comment by MGelbana-Incorta** [comment]
I restored the original formatting.

**Comment by pitbulk** [comment]
I still find places where there are spaces for example [here](https://github.com/MGelbana-Incorta/onelogin-java-sdk/blob/f5fb253222dcd86a35c4c4e6c85a711f9c40b303/src/main/java/com/onelogin/sdk/conn/Client.java#L341).

Before merging this PR, I will need to verify that this change does not break anything.

**Comment by MGelbana-Incorta** [comment]
I applied the requested formatting.

I committed test cases to make sure nothing breaks. But there were none before I started contributing so I don't know how can we verify the behaviour of earlier code.

**Comment by pitbulk** [comment]
I will test using directly a OneLogin instance

**Pull request merged**